autoload/ale/handlers.vim

```diff
@@ -116,6 +114,45 @@ function! ale#handlers#HandlePEP8Format(buffer, lines) abort
     return l:output
 endfunction
 
+
+function! ale#handlers#HandleFlake8Format(buffer, lines) abort
+    " Matches patterns line the following:
+    "
+    " stdin:6:6: E111 indentation is not a multiple of four
+    "
+    " Uses g:ale_python_flake8_error_codes to decide the severity of the
+    " message
+    let l:pattern = '^' . s:path_pattern . ':\(\d\+\):\?\(\d\+\)\?: \(\([[:alpha:]]\)[[:alnum:]]\+\) \(.*\)$'
+    let l:output = []
+
+    for l:line in a:lines
+        let l:match = matchlist(l:line, l:pattern)
+
+        if len(l:match) == 0
+            continue
+        endif
+
+        let l:code = l:match[3]
+        if (l:code ==# 'W291' || l:code ==# 'W293')
+                    \ && !g:ale_warn_about_trailing_whitespace
+            " Skip warnings for trailing whitespace if the option is off.
+            continue
+        endif
+
+        call add(l:output, {
+        \   'bufnr': a:buffer,
+        \   'lnum': l:match[1] + 0,
+        \   'vcol': 0,
+        \   'col': l:match[2] + 0,
+        \   'text': l:code . ': ' . l:match[5],
+        \   'type': index(g:ale_python_flake8_error_codes , l:code) == -1 ? 'W' : 'E',
```

Comment:
The distinction between errors and warnings I did here is not very opinion based: Everything that will keep the code from running correctly is an error, everything else is a warning. I would argue that that is the most sensible distinction, because that means you can't disagree with errors and need to fix them, but you might disagree with warnings and ignore them.

> Reading all of the F errors above, I would call them errors, not warnings.

You might have misunderstood my code – the list contains everything that is an error. Everything else is a warning. So all these F errors are actually classified as errors.

> I also think we should look at the start of the error code to distinguish between warnings or errors, so some things will be handled correctly by default.

That doesn't work. In Flake8 the letter at the start mostly informs you where the error is from. All the `E` messages are generated by `pycodestyle` (former `pep8`), so are usually style problems, not errors.

> We should probably provide no codes by default.

I strongly disagree with that, as that just provides a useless default configuration. Message examples:
* E112 	expected an indented block (this a syntax error)
* E121 	continuation line under-indented for hanging indent
* E201 	whitespace after ‘(‘
* E999 Generic Syntax Error
* F823 	local variable name ... referenced before assignment (this is always a runtime error)
* F701 	a break statement outside of a while or for loop (this is a syntax error)
* F401 	module imported but unused

So are E and F messages errors or warnings?

> I'd write these changes into the HandlePEP8Format function above, change the global variable name to g:ale_python_error_codes, and document it in the Python documentation. 

You seem to be assuming that all python tools share these error codes? That is not correct. `flake8` inherits `pycodestyle`'s (former `pep8`) list and extends it, but pylint completely does its own thing.

> I reckon we should create a Dictionary instead of a List, and the values in the dictionaries should be the error types. That way, you could map some things to errors, other things to warnings, and then later some things to style warnings. ('S') That will be useful for configuring extra flake8 plugins.

I don't quite see the point of this – what is the difference between a warning and a style warning?